include/solvers/gd_navier_stokes.h

```diff
@@ -33,10 +33,16 @@
 #include <deal.II/lac/trilinos_precondition.h>
 #include <deal.II/lac/trilinos_solver.h>
 
-
-
 using namespace dealii;
 
+DeclException3(SolidWarning,
```

Comment:
You can declare the exception once somewhere in core for example. This will prevent you from having to redeclare it for every solver. You can make a  core/exceptions.h if you want even.